src/main/java/org/folio/service/pieces/PieceStorageService.java

```diff
@@ -110,10 +119,26 @@ public Future<List<Piece>> getPiecesByHoldingId(String holdingId, RequestContext
   }
 
   public Future<PieceCollection> getPieces(int limit, int offset, String query, RequestContext requestContext) {
-    RequestEntry requestEntry = new RequestEntry(PIECE_STORAGE_ENDPOINT).withQuery(query)
-      .withOffset(offset)
-      .withLimit(limit);
-    return restClient.get(requestEntry, PieceCollection.class, requestContext);
+    var requestEntry = new RequestEntry(PIECE_STORAGE_ENDPOINT).withQuery(query).withOffset(offset).withLimit(limit);
+    return restClient.get(requestEntry, PieceCollection.class, requestContext)
+      .compose(piecesCollection -> filterPiecesByUserTenantsIfNecessary(piecesCollection.getPieces(), requestContext))
+      .map(pieces -> new PieceCollection().withPieces(pieces).withTotalRecords(pieces.size()));
```

Comment:
should be piecesCollection.getTotal() instead of pieces.size(). It breaks pagination but at least for non ECS envs pagination will work

Reply:
Good point. Pagination can be still kept valid in case we implement a mechanism to retry GET requests until we get desired number of pieces with correct offset parameters. Could be some work, or should we keep this as is?

Reply:
Probably let leave as is to make solution simple, we do not have pagination for pieces for now, and in case it would be some integration in ECS mode to retrieve pieces - they just need to have an affiliation in each tenant to bypass this, because there is no other clear way here.
Just need to change pieces.size() to total